### PR TITLE
[MI-3255]: Added svg icon and Tooltip

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
## Summary

- Added svg icon to options button existing config table buttons.
- Added tooltip to buttons

### CODE REVIEW

File: webapp/package-lock.json
Comment (raghavaggarwal2308) [inline webapp/package-lock.json:1]: Revert these changes

File: webapp/src/containers/components/modals/actionModal.tsx [hunk 1]
```diff
@@ -81,14 +88,14 @@ function ViewActionsModal(props: Props) {
...
-                    >{'Save changes'}</Button>
+                    >{'Save changes'}</Button> */}
                 </Modal.Footer>
             </Modal>
         </>
     );
 }
 
-export default ViewActionsModal;
+export default ActionModal;
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/actionModal.tsx]: Add EOF

File: webapp/src/containers/components/modals/deleteModal.tsx [hunk 2]
```diff
@@ -46,4 +48,4 @@ function DeleteConfigModal(props: Props) {
     );
 }
 
-export default DeleteConfigModal;
\ No newline at end of file
+export default DeleteModal;
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/deleteModal.tsx]: Add EOF

File: webapp/src/containers/components/modals/styles.css [hunk 3]
```diff
@@ -1,4 +1,110 @@
...
+
+.tale th {
+  text-align: center;
+  vertical-align: bottom;
+} 
+
+.table {
+  width: 100%;
+}
+.table thead tr th td{
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/styles.css]:
```suggestion

.table thead tr th td{
```

File: webapp/src/containers/components/modals/styles.css [hunk 4]
```diff
@@ -1,4 +1,110 @@
...
+}
+
+table.table tbody tr td,
+table.table thead tr th,
+table.table thead {
+  border: 2px solid #ddd;
+  vertical-align: middle;
+}
+  
+.table td {
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/styles.css]: Give a parent class for these. Check everywhere

File: webapp/src/containers/components/tables/existingConfigTable.tsx [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
 import Button from 'react-bootstrap/Button';
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
+import Tooltip from 'react-bootstrap/Tooltip';
 
 import './styles.css';
 
-import ViewActionsModal from '../modals/viewActionsModal';
-import DeleteConfigModal from '../modals/deleteConfigModal';
+import {Config} from '../../../types/plugin/common';
+
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/tables/existingConfigTable.tsx]: Fix import order in this file

File: webapp/src/containers/components/tables/existingConfigTable.tsx [hunk 6]
```diff
@@ -32,164 +36,476 @@ type Props = {
...
+                                        <OverlayTrigger
+                                            placement='top'
+                                            overlay={<Tooltip>{'View actions'}</Tooltip>}
+                                        >
                                             <Button
-                                                variant='primary'
                                                 onClick={handleView}
                                             >
-                                                {'View'}
+                                                <svg
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/tables/existingConfigTable.tsx]: Create a separate file for these svgs and import them here.

File: webapp/src/containers/components/tables/styles.css [hunk 7]
```diff
@@ -1,27 +1,93 @@
...
+
+.existing-config-table.btn-group .btn.btn-default {
+    background: none;
+    border-color: transparent;
+    border: transparent;
+}
+
+.existing-config-table .btn-group.btn.btn-default:hover, .btn-group .btn.btn-default:focus {
+    background: transparent;
+    color: #000000;
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/tables/styles.css]:
```suggestion
    color: #000;
```

File: webapp/src/containers/components/tables/styles.css [hunk 8]
```diff
@@ -1,27 +1,93 @@
...
+    text-overflow: ellipsis;
+    white-space: nowrap;
+}
+
+.option {
+    width: 10%;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+}
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/tables/styles.css]: Add EOF

File: webapp/src/containers/components/modals/deleteModal.tsx [hunk 9]
```diff
@@ -1,14 +1,17 @@
 import React, {useEffect, useState} from 'react';
+
 import Button from 'react-bootstrap/Button';
 import Modal from 'react-bootstrap/Modal';
+
 import './styles.css';
 
 interface Props {
     visible: boolean;
     setVis: React.Dispatch<React.SetStateAction<boolean>>;
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/deleteModal.tsx]:
```suggestion
    setVisibilty: React.Dispatch<React.SetStateAction<boolean>>;
```

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/containers/components/tables/existingConfigTable.tsx (hunk 6 above)
Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/tables/existingConfigTable.tsx]: Will fix this in the next PR

File: webapp/src/containers/components/modals/styles.css (hunk 4 above)
Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/styles.css]: Fixed in the next PRs

File: webapp/src/containers/components/tables/existingConfigTable.tsx [hunk 10]
```diff
@@ -61,14 +60,13 @@ const ExistingConfigTable = ({label, helpText}: Props) => {
             {viewVisible &&
                 <ActionModal
                     visible={viewVisible}
-                    setVis={setViewVisible}
+                    setVisible={setViewVisible}
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/tables/existingConfigTable.tsx]:
```suggestion
                    setVisibility={setViewVisibility}
```

File: webapp/src/containers/components/modals/deleteModal.tsx [hunk 11]
```diff
@@ -20,7 +17,7 @@ function DeleteModal(props: Props) {
 
     const handleClose = () => {
         setShow(false);
-        props.setVisibility(false);
+        props.setVis(false);
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/deleteModal.tsx]: Why this change

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/containers/components/modals/actionModal.tsx [hunk 12]
```diff
@@ -77,10 +87,10 @@ const ViewActionsModal = (props: Props) => {
...
                     onClick={handleClose}
                 >{'Close'}</Button>
-                <Button
-                    variant='primary'
-                    disabled={true}
-                >{'Save changes'}</Button>
+                {/* <Button
+                        variant='primary'
+                        disabled={true}
+                    >{'Save changes'}</Button> */}
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/actionModal.tsx]: remove if not needed or add TODO comment

File: webapp/src/containers/components/modals/deleteModal.tsx [hunk 13]
```diff
@@ -0,0 +1,49 @@
+import React, {useEffect, useState} from 'react';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+import './styles.css';
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/deleteModal.tsx]:
```suggestion

import './styles.css';
```

File: webapp/src/containers/components/modals/deleteModal.tsx [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+interface Props {
+    visible: boolean;
+    setVis: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+function DeleteConfigModal(props: Props) {
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/deleteModal.tsx]: use `type` and arrow function

File: webapp/src/containers/components/tables/styles.css [hunk 15]
```diff
@@ -1,27 +1,93 @@
...
+    padding-left: 15px;
+}
+
+.name {
+    font-weight: 600;
+    font-size: 14px;
+    margin-bottom: 1em;
+}
+
+.teamName {
```
Comment (avas27JTG) [inline webapp/src/containers/components/tables/styles.css]: don't use `camelCase` for CSS classes always use `kebab-case`

File: webapp/src/containers/components/tables/styles.css [hunk 16]
```diff
@@ -1,27 +1,93 @@
...
+}
+
+.message {
+    width: 55%;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+}
+
+.includeGuests {
```
Comment (avas27JTG) [inline webapp/src/containers/components/tables/styles.css]: same

File: webapp/src/utils/index.ts [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+
+const getBaseUrls = (): {pluginApiBaseUrl: string; mattermostApiBaseUrl: string} => {
+    const url = new URL(window.location.href);
+    const baseUrl = `${url.protocol}//${url.host}`;
+    const pluginUrl = `${baseUrl}/plugins/${id}`;
+    const pluginApiBaseUrl = `${pluginUrl}/api/v1`;
+    const mattermostApiBaseUrl = `${baseUrl}/api/v4`;
+
+    return {pluginApiBaseUrl, mattermostApiBaseUrl};
+};
```
Comment (avas27JTG) [inline webapp/src/utils/index.ts]: don't use `window` it can be vulnerable to attacks, you can see https://github.com/Brightscout/mattermost-plugin-template/blob/bfef1df581b40461702d166d7007f40509cb76a3/webapp/src/utils/index.ts#L6

File: webapp/src/containers/components/modals/deleteModal.tsx (hunk 11 above)
Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/deleteModal.tsx]: fixed the variable name to setVisibility

File: webapp/package-lock.json
Comment (raghavaggarwal2308) [inline webapp/package-lock.json:1]: revert changes of this file

File: webapp/src/containers/components/modals/actionModal.tsx [hunk 18]
```diff
@@ -60,15 +65,18 @@ const ViewActionsModal = (props: Props) => {
                             <td>{'Import'}</td>
                             <td>{'channel1, channel2, channel3'}</td>
                             <td>{'Welcome to your new team mate !'}</td>
+                        </tr>
+                        <tr>
                             <td>{'Automatic'}</td>
                             <td>{'Export'}</td>
                             <td>{'channel1, channel2'}</td>
-                            <td>{'Welcome to your new team mate!'}</td>
+                            <td>{'Welcome to your new team mate !'}</td>
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/actionModal.tsx]:
```suggestion
                            <td>{'Welcome to your new team mate!'}</td>
```

File: webapp/src/containers/components/modals/actionModal.tsx [hunk 19]
```diff
@@ -60,15 +65,18 @@ const ViewActionsModal = (props: Props) => {
...
                         </tr>
                         <tr>
                             <td>{'Button'}</td>
                             <td>{'Deport'}</td>
                             <td>{'channel1, channel3'}</td>
+                            <td>{'Welcome to your new team mate !'}</td>
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/actionModal.tsx]:
```suggestion
                            <td>{'Welcome to your new team mate!'}</td>
```

File: webapp/src/containers/components/modals/configModal.tsx [hunk 20]
```diff
@@ -0,0 +1,454 @@
...
+
+type Props = {
+    visible: boolean;
+    setVisibility: React.Dispatch<React.SetStateAction<boolean>>;
+    config: Config | null;
+}
+
+function ConfigModal(props: Props) {
+    const [show, setShow] = useState(false);
+    const [actionVisible, setActionVisible] = useState(false);
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]:
```suggestion
    const [isActionVisible, setIsActionVisible] = useState(false);
```
Fix everywhere

File: webapp/src/containers/components/modals/configModal.tsx [hunk 21]
```diff
@@ -0,0 +1,454 @@
...
+            setActionVisible(false);
+            setConfigVisible(true);
+        } else if (deleteVisible) {
+            setDeleteVisible(false);
+            setConfigVisible(true);
+        } else {
+            setShow(false);
+            props.setVisibility(false);
+        }
+    };
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]:
```suggestion
    };
    
```

File: webapp/src/containers/components/modals/configModal.tsx [hunk 22]
```diff
@@ -0,0 +1,454 @@
...
+    const handleActions = () => {
+        setActionVisible(true);
+        setConfigVisible(false);
+    };
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]:
```suggestion
    };
    
```

File: webapp/src/containers/components/modals/configModal.tsx [hunk 23]
```diff
@@ -0,0 +1,454 @@
...
+    const handleDelete = () => {
+        setDeleteVisible(true);
+        setConfigVisible(false);
+    };
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]:
```suggestion
    };
    
```

File: webapp/src/containers/components/modals/configModal.tsx (hunk 21 above)
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]:
```suggestion
    const handleClose = () => {
        if (actionVisible) {
            setActionVisible(false);
            setConfigVisible(true);
            return;
        } else if (deleteVisible) {
            setDeleteVisible(false);
            setConfigVisible(true);
            return;
        }
        
        setShow(false);
        props.setVisibility(false);
    };
```

File: webapp/src/containers/components/modals/configModal.tsx [hunk 24]
```diff
@@ -0,0 +1,454 @@
...
+// eslint-disable-next-line import/no-unresolved
+import {Config} from 'types/plugin/common';
+
+type Props = {
+    visible: boolean;
+    setVisibility: React.Dispatch<React.SetStateAction<boolean>>;
+    config: Config | null;
+}
+
+function ConfigModal(props: Props) {
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]: Destructure the props

File: webapp/src/containers/components/modals/configModal.tsx [hunk 25]
```diff
@@ -0,0 +1,454 @@
...
+                                                        <polygon points='18 2 22 6 12 16 8 16 8 12 18 2'/>
+                                                    </svg>
+                                                </Button>
+                                            </OverlayTrigger>
+                                            <OverlayTrigger
+                                                placement='top'
+                                                overlay={<Tooltip>{'Delete action'}</Tooltip>}
+                                            >
+                                                <Button onClick={handleDelete}>
+                                                    <svg
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]: I think we created a separate file for svgs? If not, create it now

File: webapp/src/containers/components/modals/configModal.tsx [hunk 26]
```diff
@@ -0,0 +1,454 @@
...
+                        {'Save changes'}
+                    </Button>}
+                    {actionVisible &&
+                    <Button
+                        variant='primary'
+                        onClick={handleClose}
+                    >
+                        {'Add action'}
+                    </Button>}
+                    {deleteVisible &&
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]: I think you can combine it with the conditional on line 420

File: webapp/src/containers/components/modals/deleteModal.tsx [hunk 27]
```diff
@@ -0,0 +1,51 @@
+import React, {useEffect, useState} from 'react';
+
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+
+import './styles.css';
+
+type Props = {
+    visible: boolean;
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/deleteModal.tsx]:
```suggestion
    visibility: boolean;
```
Fix this in all the PRs. it should be visibility and setVisibility. Every var name should make proper sense. So that it clears the use of the code if any other dev is working on it. The names you have given also clear the meaning but it doesn't sounds good.

File: webapp/src/containers/components/modals/deleteModal.tsx [hunk 28]
```diff
@@ -0,0 +1,51 @@
...
+    setVisibility: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const DeleteConfigModal = (props: Props) => {
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/deleteModal.tsx]: destructure props

File: webapp/src/containers/components/modals/deleteModal.tsx [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+    const [show, setShow] = useState(false);
+
+    useEffect(() => {
+        setShow(props.visible);
+    }, [props.visible]);
+
+    const handleClose = () => {
+        setShow(false);
+        props.setVisibility(false);
+    };
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/deleteModal.tsx]:
```suggestion
    };
    
```

File: webapp/src/containers/components/modals/deleteModal.tsx (hunk 27 above)
Comment (raghavaggarwal2308) [reply webapp/src/containers/components/modals/deleteModal.tsx]: Also, I think we removed this variable right?

File: webapp/src/containers/components/modals/styles.css [hunk 30]
```diff
@@ -1,4 +1,111 @@
...
+.options .btn.btn-primary {
+  background: transparent;
+}
+
+.radio {
+  margin-top: 0;
+  margin-bottom: 0;
+}
+
+.listTable {
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/styles.css]: don't use camel case names

File: webapp/src/containers/components/tables/existingConfigTable.tsx [hunk 31]
```diff
@@ -1,15 +1,18 @@
 import React, {useState} from 'react';
 
-import FormGroup from 'react-bootstrap/FormGroup';
-import Col from 'react-bootstrap/Col';
 import Table from 'react-bootstrap/Table';
 import ButtonGroup from 'react-bootstrap/ButtonGroup';
 import Button from 'react-bootstrap/Button';
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
+import Tooltip from 'react-bootstrap/Tooltip';
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/tables/existingConfigTable.tsx]: group in single import

File: webapp/src/containers/components/tables/existingConfigTable.tsx [hunk 32]
```diff
@@ -32,159 +35,475 @@ type Props = {
...
+                            <th className='include-guests'>{'Include Guests'}</th>
+                            <th className='option'>{'Options'}</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        <tr>
+                            <td className='team-name'>{'Standup'}</td>
+                            <td className='delay'>{'4'}</td>
+                            <td className='message'>
+                                {'Hello to standup group sdhfvk.'}
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/tables/existingConfigTable.tsx]: Add a todo above this to use proper names

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/containers/components/modals/actionModal.tsx [hunk 33]
```diff
@@ -34,18 +28,23 @@ const ViewActionsModal = (props: Props) => {
 
             <Modal.Body>
                 <Form>
-                    <Form.Group>
+                    <Form.Group className='form-group'>
                         <Form.Label>{'Attachment Message'}</Form.Label>
                         <Form.Control
                             type='long-text'
+                            placeholder='This is a example attachment message'
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/actionModal.tsx]:
```suggestion
                            placeholder='This is an example attachment message'
```

File: webapp/src/containers/components/modals/actionModal.tsx [hunk 34]
```diff
@@ -59,7 +58,9 @@ const ViewActionsModal = (props: Props) => {
                             <td>{'Button'}</td>
                             <td>{'Import'}</td>
                             <td>{'channel1, channel2, channel3'}</td>
-                            <td>{'Welcome to your new team mate !'}</td>
+                            <td>{'Welcome to your new team mate!'}</td>
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/actionModal.tsx]:
```suggestion
                            <td>{'Welcome to your new team!'}</td>
```

File: webapp/src/containers/components/modals/actionModal.tsx [hunk 35]
```diff
@@ -69,6 +70,7 @@ const ViewActionsModal = (props: Props) => {
                             <td>{'Button'}</td>
                             <td>{'Deport'}</td>
                             <td>{'channel1, channel3'}</td>
+                            <td>{'Welcome to your new team mate!'}</td>
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/actionModal.tsx]:
```suggestion
                            <td>{'Welcome to your new team!'}</td>
```

File: webapp/src/containers/components/modals/deleteModal.tsx [hunk 36]
```diff
@@ -0,0 +1,45 @@
...
+
+type Props = {
+    visibility: boolean;
+    setVisibility: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const DeleteConfigModal = ({visibility, setVisibility}: Props) => {
+    const handleClose = () => {
+        setVisibility(false);
+    };
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/deleteModal.tsx]:
```suggestion
    const handleClose = () => setVisibility(false);
```

File: webapp/src/containers/components/tables/styles.css [hunk 37]
```diff
@@ -1,27 +1,93 @@
...
+
+.existing-config-table.btn-group .btn.btn-default {
+    background: none;
+    border-color: transparent;
+    border: transparent;
+}
+
+.existing-config-table .btn-group.btn.btn-default:hover, .btn-group .btn.btn-default:focus {
+    background: transparent;
+    color: #000;
```
Comment (avas27JTG) [inline webapp/src/containers/components/tables/styles.css]:
```suggestion
    color: #000000;
```

File: webapp/src/containers/components/tables/styles.css [hunk 38]
```diff
@@ -1,27 +1,93 @@
...
+}
+
+.add-config-btn {
+    margin: 1em 0 0 0;
```
Comment (avas27JTG) [inline webapp/src/containers/components/tables/styles.css]: use consistent CSS unit somewhere `%`, `em`, `px` all are used?

File: webapp/src/types/plugin/common.d.ts [hunk 39]
```diff
@@ -0,0 +1,4 @@
+export type Config = {
+    ConfigValues: boolean,
+}
+
```
Comment (avas27JTG) [inline webapp/src/types/plugin/common.d.ts]: why do we need explicit export when this issue is already resolved?

File: webapp/src/containers/components/modals/configModal.tsx (hunk 25 above)
Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/configModal.tsx]: Created in #14

File: webapp/src/containers/components/modals/configModal.tsx (hunk 26 above)
Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/configModal.tsx]: combining them in one conditional will force us to add extra css, also we have removed this button in next PR

**Review by raghavaggarwal2308 — APPROVED** [review]

**Review by avas27JTG — APPROVED** [review]
